app/src/main/java/com/zulip/android/activities/MessageListFragment.java

```diff
@@ -398,70 +401,101 @@ private void editMessage(final Message message, final int position) {
-                @Override
-                public void onClick(View view) {
-                    dialog.cancel();
-                    InputMethodManager imm = (InputMethodManager) getActivity().getSystemService(Context.INPUT_METHOD_SERVICE);
-                    imm.hideSoftInputFromWindow(view.getWindowToken(), 0);
-                    //Start a progress dialog indicating editing message
-                    final ProgressDialog progress = new ProgressDialog(getActivity());
-                    progress.setCancelable(false);
-                    progress.setMessage(app.getString(R.string.editing_message));
-                    progress.show();
-                    final String editedMessageContent =
-                            dialogMessageEditText.getText().toString().length() == 0 ? getString(R.string.default_delete_text) : dialogMessageEditText.getText().toString();
-
-                    app.getZulipServices()
-                            .editMessage(String.valueOf(message.getID()), editedMessageContent)
-                            .enqueue(new Callback<EditResponse>() {
-                                @Override
-                                public void onResponse(Call<EditResponse> call, Response<EditResponse> response) {
-                                    if (response.isSuccessful()) {
-                                        message.setContent(editedMessageContent);
-                                        message.setFormattedContent(editedMessageContent);
-                                        adapter.notifyItemChanged(position);
-                                        progress.dismiss();
-                                        Toast.makeText(getActivity(), R.string.message_edited, Toast.LENGTH_SHORT).show();
-                                    } else {
-                                        progress.dismiss();
-                                        Toast.makeText(getActivity(), R.string.message_edit_failed, Toast.LENGTH_SHORT).show();
-                                    }
-                                }
+            final CommonProgressDialog commonProgressDialog = new CommonProgressDialog(getContext());
+            commonProgressDialog.showWithMessage(getString(R.string.fetch_edit_message));
+            app.getZulipServices()
+                    .fetchRawMessage(message.getID())
+                    .enqueue(new Callback<RawMessageResponse>() {
+                        @Override
+                        public void onResponse(Call<RawMessageResponse> call, Response<RawMessageResponse> response) {
+                            RawMessageResponse messageResponse = response.body();
+                            if (!messageResponse.getResult().equalsIgnoreCase("success")) {
+                                Toast.makeText(getContext(), (messageResponse.getMsg() == null || TextUtils.isEmpty(messageResponse.getMsg())) ? getString(R.string.message_edit_failed) :
```

Comment:
I think textUtils already checks for str == null, hence we can omit `messageResponse.getMsg() == null || `